### PR TITLE
Reenable unified instruction set logic

### PR DESCRIPTION
**Pull request by davidwrighton**
Revert #33901 to reenable #33730. The issue was that the runtime was incorrectly disabling use in the jit compiler for all hardware intrinsics on 64 bit platforms at runtime. Fixed by calling `Set64BitInstructionSetVariants` before `EnsureValidInstructionSetSupport`.

### CODE REVIEW

**Comment by davidwrighton** [comment]
@dotnet/jit-contrib Note, this changes the jit interface

**Comment by sandreenko** [comment]
@davidwrighton azp run is useless now, sew https://github.com/dotnet/runtime/issues/32777

File: src/coreclr/src/inc/corinfoinstructionset.h [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+#endif // TARGET_ARM64
+#ifdef TARGET_AMD64
+    InstructionSet_SSE=1,
+    InstructionSet_SSE2=2,
+    InstructionSet_SSE3=3,
+    InstructionSet_SSSE3=4,
+    InstructionSet_SSE41=5,
+    InstructionSet_SSE42=6,
+    InstructionSet_AVX=7,
+    InstructionSet_AVX2=8,
```
Comment (lpereira) [inline src/coreclr/src/inc/corinfoinstructionset.h]: Not needed at this point (still plenty of room in the bitmask), but for instruction sets that are implied by other instruction sets, a better strategy could be used here to pack these bits.  For instance, for x86 SIMD, a "SIMD level" thing could be used (e.g. `InstructionSet_SIMD_Level >= InstructionSet_SSE42` to test if SSE 4.2 is supported, with `InstructionSet_SIMD_Level` being in this case a 3-bit mask).

Comment (davidwrighton) [reply src/coreclr/src/inc/corinfoinstructionset.h]: Agreed, one thing to note is that the representation as a 64bit bitmask is an internal implementation detail, and we can change it fairly easily when we reach to having more than 62 instruction sets on a specific architecture. I believe that this scheme should last for at least 5 more years, which is long enough I don't care to plan further ahead.

**Review by fadimounir — APPROVED** [review]
LGTM. I did not review the actual contents from the original PR that was reverted. I just looked at the `EnsureValidInstructionSetSupport` fix, and the JIT interface guid change

**Pull request merged**